### PR TITLE
Update docs helper. Automate generation of 'Merging lists of dictiona…

### PR DESCRIPTION
**Pull request by vbotka**
Update docs helper. Automate generation of 'Merging lists of dictionaries'.

* Updated helper/lists_mergeby/playbook.yml, list of examples and
  templates. See playbook.yml on how to create *.out files, test
  examples and generate the REST file
  filter_guide_abstract_informations_merging_lists_of_dictionaries.rst
* Generated REST file copied to directory rst
* Simplified examples. The common lists are published only once. Only
  the expressions are published instead of the whole task.
* To change the content of the section 'Merging lists of dictionaries'
  update template
  filter_guide_abstract_informations_merging_lists_of_dictionaries.rst.j2
  and run the playbook.
* Deleted rst/examples/lists_mergeby. Not needed anymore.

##### SUMMARY

Automate the generation of the section 'Merging lists of dictionaries'. The goal is to make updating and maintenance easier. The procedure is easily extendable and reusable.

**How-to update the section**

In the directory docs/docsite/helper/lists_mergeby:

* Create new or update the examples example-XXX.yml and the directories example-XXX_var
* See the content of the directories example-XXX_vars and the published expression, e.g. list3.yml, in particular
* Update the list of the examples examples.yml if needed
* Run the playbook and create the output files *.out. Run ``ansible-playbook playbook.yml -e examples=true``
* Optionally, create and review the REST file examples_all.rst. Run ``ansible-playbook playbook.yml -e examples_all=true``
* To update the section make changes in the template filter_guide_abstract_informations_merging_lists_of_dictionaries.rst.j2
* Create and proofread the REST file. Run ``ansible-playbook playbook.yml -e merging_lists_of_dictionaries=true``
* Copy the REST file to the directory rst

##### ISSUE TYPE

- Docs Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by vbotka** [comment]
Despite the fact, that changes seem extensive there are no substantial modifications in the content of the section. There are only two changes in general: 1) The common lists (list1 and list2) are published only once in front of the examples and 2) Only the expression (list3) is published instead of the whole task.

File: docs/docsite/helper/lists_mergeby/filter_guide_abstract_informations_merging_lists_of_dictionaries.rst.j2 [hunk 1]
```diff
@@ -0,0 +1,57 @@
+Merging lists of dictionaries
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you have two or more lists of dictionaries and want to combine them into a list of merged dictionaries, where the dictionaries are merged by an attribute, you can use the ``lists_mergeby`` filter.
+
+.. note:: The output of the examples in this section use the YAML callback plugin. Quoting: "Ansible output that can be quite a bit easier to read than the default JSON formatting." See :ref:`the documentation for the community.general.yaml callback plugin <ansible_collections.community.general.yaml_callback>`.
+
+Let us use the lists below in the following examples
```
Comment (felixfontein) [inline docs/docsite/helper/lists_mergeby/filter_guide_abstract_informations_merging_lists_of_dictionaries.rst.j2:8]:
```suggestion
Let us use the lists below in the following examples:
```

**Review by felixfontein — COMMENTED** [review]
Besides this, looks good!

**Comment by vbotka** [comment]
> Besides this, looks good!

Thank you for the review! By mistake, I set the *bug* label. I'm afraid that I'm not able to remove it on my own.  Sorry for the noise.

**Comment by felixfontein** [comment]
No problem, I've adjusted it. Can you regenerate the RST file with the updated template?

**Comment by vbotka** [comment]
Done. I regenerated the RST file with the updated template.

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@vbotka thanks a lot for this improvement!

**Pull request merged**